### PR TITLE
refactor: tidy `actions` submodules and improve API readability

### PR DESCRIPTION
**Pull request by Akmadan23**
Discussed in #2583

### CODE REVIEW

File: lua/nvim-tree/api.lua [hunk 1]
```diff
@@ -205,54 +189,59 @@ Api.node.open.tab = wrap_node(open_or_expand_or_dir_up "tabnew")
...
+local live_filter = require "nvim-tree.live-filter"
+
+Api.live_filter.start = wrap(live_filter.start_filtering)
+Api.live_filter.clear = wrap(live_filter.clear_filter)
+
+local marks = require "nvim-tree.marks"
+local marks_navigation = require "nvim-tree.marks.navigation"
+local marks_bulk_delete = require "nvim-tree.marks.bulk-delete"
+local marks_bulk_trash = require "nvim-tree.marks.bulk-trash"
+local marks_bulk_move = require "nvim-tree.marks.bulk-move"
```
Comment (Akmadan23) [inline lua/nvim-tree/api.lua:227]: I tried to add the various `init.lua`s in the `marks` module but dependency loops start to come up, so this is best solution I could find.

**Review by Akmadan23 — COMMENTED** [review]
The move of `actions/tree-modifiers` to `actions/tree/modifiers` is just a proposal, I think it makes sense to put it there but I can revert this change if it's better not to do that.
I would also move `actions/reloaders/reloaders.lua` to `actions/reloaders/init.lua` or even `actions/reloaders.lua`, as long as it's not a problem...

File: lua/nvim-tree/api.lua [hunk 2]
```diff
@@ -205,54 +189,59 @@ Api.node.open.tab = wrap_node(open_or_expand_or_dir_up "tabnew")
...
+Api.node.navigate.diagnostics.next = wrap_node(actions.moves.item.fn { where = "next", what = "diag" })
+Api.node.navigate.diagnostics.prev = wrap_node(actions.moves.item.fn { where = "prev", what = "diag" })
+Api.node.navigate.opened.next = wrap_node(actions.moves.item.fn { where = "next", what = "opened" })
+Api.node.navigate.opened.prev = wrap_node(actions.moves.item.fn { where = "prev", what = "opened" })
+
+Api.git.reload = wrap(actions.reloaders.reload_git)
+
+local events = require "nvim-tree.events"
+
+Api.events.subscribe = wrap(events.subscribe)
```
Comment (alex-courtis) [inline lua/nvim-tree/api.lua:215]: We can't wrap this one - we alllow users to subscribe before setup is called.

File: lua/nvim-tree/api.lua [hunk 3]
```diff
@@ -205,54 +189,59 @@ Api.node.open.tab = wrap_node(open_or_expand_or_dir_up "tabnew")
...
+Api.marks.bulk.move = wrap(marks_bulk_move.bulk_move)
+Api.marks.navigate.next = wrap(marks_navigation.next)
+Api.marks.navigate.prev = wrap(marks_navigation.prev)
+Api.marks.navigate.select = wrap(marks_navigation.select)
+
+local keymap = require "nvim-tree.keymap"
+
+Api.config.mappings.get_keymap = wrap(keymap.get_keymap)
+Api.config.mappings.get_keymap_default = wrap(keymap.get_keymap_default)
+Api.config.mappings.default_on_attach = wrap(keymap.default_on_attach)
```
Comment (alex-courtis) [inline lua/nvim-tree/api.lua:244]: No wrap for this one either; we let the users call this for before setup, even though it's an odd use case.

File: lua/nvim-tree/api.lua [hunk 4]
```diff
@@ -205,54 +189,59 @@ Api.node.open.tab = wrap_node(open_or_expand_or_dir_up "tabnew")
...
+Api.marks.toggle = wrap_node(marks.toggle_mark)
+Api.marks.clear = wrap(marks.clear_marks)
+Api.marks.bulk.delete = wrap(marks_bulk_delete.bulk_delete)
+Api.marks.bulk.trash = wrap(marks_bulk_trash.bulk_trash)
+Api.marks.bulk.move = wrap(marks_bulk_move.bulk_move)
+Api.marks.navigate.next = wrap(marks_navigation.next)
+Api.marks.navigate.prev = wrap(marks_navigation.prev)
+Api.marks.navigate.select = wrap(marks_navigation.select)
+
+local keymap = require "nvim-tree.keymap"
```
Comment (alex-courtis) [inline lua/nvim-tree/api.lua:240]: Could we move these requires to the top, or is order important here?

**Comment by alex-courtis** [comment]
> The move of `actions/tree-modifiers` to `actions/tree/modifiers` is just a proposal, I think it makes sense to put it there but I can revert this change if it's better not to do that. I would also move `actions/reloaders/reloaders.lua` to `actions/reloaders/init.lua` or even `actions/reloaders.lua`, as long as it's not a problem...

That's great - it is a hierarchy which makes sense.

File: lua/nvim-tree/api.lua (hunk 1 above)
Comment (alex-courtis) [reply lua/nvim-tree/api.lua:227]: Marks are not quite like the other actions. They need a refactor / tidy but not today.

**Review by alex-courtis — CHANGES_REQUESTED** [review]
This is incredible! Api is finally readable. There's very little more we can do for now...

Now you have me thinking of ways we could get rid of wrap...

File: lua/nvim-tree/api.lua (hunk 4 above)
Comment (Akmadan23) [reply lua/nvim-tree/api.lua:240]: I put the definitions just before the references to easily trace back which module each variable refers to, of course we can move them to the top.

File: lua/nvim-tree/actions/fs/copy-paste.lua [hunk 5]
```diff
@@ -5,7 +5,7 @@ local core = require "nvim-tree.core"
 local events = require "nvim-tree.events"
 local notify = require "nvim-tree.notify"
 local renderer = require "nvim-tree.renderer"
-local reloaders = require "nvim-tree.actions.reloaders.reloaders"
+local reloaders = require "nvim-tree.actions.reloaders"
```
Comment (alex-courtis) [inline lua/nvim-tree/actions/fs/copy-paste.lua:8]: Nice

File: lua/nvim-tree/actions/init.lua [hunk 6]
```diff
@@ -1,18 +1,18 @@
...
-  require("nvim-tree.actions.fs.remove-file").setup(opts)
-  require("nvim-tree.actions.fs.copy-paste").setup(opts)
-  require("nvim-tree.actions.tree-modifiers.expand-all").setup(opts)
-  require("nvim-tree.actions.tree.find-file").setup(opts)
-  require("nvim-tree.actions.tree.open").setup(opts)
-  require("nvim-tree.actions.tree.toggle").setup(opts)
+  M.fs.setup(opts)
+  M.node.setup(opts)
+  M.root.setup(opts)
+  M.tree.setup(opts)
```
Comment (alex-courtis) [inline lua/nvim-tree/actions/init.lua:15]: This `@PostConstruct` style is an unfortunate necessity for lua's "dependency injection".

I'll be mindful to add more of this as I touch files with odd instantiation mechanisms.

**Review by alex-courtis — APPROVED** [review]
Love your work. Let's get this in!

**Pull request merged**